src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2323,9 +2345,9 @@ private void LexDirectiveAndExcludedTrivia(
         {
             var directive = this.LexSingleDirective(true, true, afterFirstToken, afterNonWhitespaceOnLine, ref triviaList);
 
-            // also lex excluded stuff            
-            var branching = directive as BranchingDirectiveTriviaSyntax;
-            if (branching != null && !branching.BranchTaken)
+            // also lex excluded stuff
+            if (!afterNonWhitespaceOnLine &&
```

Comment:
>if (!afterNonWhitespaceOnLine && [](http://example.com/codeflow?start=12&length=32)

It feels strange to call `LexDirective**AndExcludedTrivia**` and then modify its code to not lex the excluded trivia. Consider reverting this change and calling ```LexSingleDirective``` instead.